libethereum/ClientBase.h

```diff
@@ -174,7 +174,9 @@ class ClientBase: public Interface
     }
 
     Block block(BlockNumber _h) const;
-
+    
+    virtual int chainId() const override;
```

Comment:
Do not repeat `virtual` keyword if `override` is used.

Reply:
Thanks for the feedback @chfast! I followed the approach used by other functions in the class e.g.:

https://github.com/ethereum/cpp-ethereum/blob/3f2ac7c4f9e1543eb4f31cbdcdf8028eb9bfe0b6/libethereum/ClientBase.h#L130

After reading more about the virtual and override keywords, it looks like neither of them are required:
* Virtual: This keyword is redundant because ClientBase::chainId() is implicitly virtual because it overrides a virtual function
* Override: This keyword isn't required, but it can be help one avoid runtime bugs since it tells the compiler to verify that there's a function in the parent class with the same signature and to raise a compiler error if that's not the case

That said, I can also see the value in keeping both keywords in the interest of code consistency. What are your thoughts?

Reply:
Yes, `override` is helpful and having both `virtual` and `override` is redundant. I'm fine with making it less consistent and leaving only `override`.
You could also remove all other `virtual` from other overridden methods.

Reply:
Thanks @gumb0, I've removed virtual from all of the ClientBase functions which override other functions.